src/utils.ts

```diff
@@ -0,0 +1,37 @@
+export function u8ArrayToBytes(array: Uint8Array) {
+  let ret = "";
+  for (let e of array) {
+    ret += String.fromCharCode(e);
+  }
+  return ret;
+}
+
+// TODO this function is a bit broken and the type can't be string
+// TODO for more info: https://github.com/near/near-sdk-js/issues/78
+export function bytesToU8Array(bytes: any): Uint8Array {
```

Comment:
Also here, typedef `Bytes` and make it `bytesToU8Array(bytes: Bytes): Uint8Array` also address this, Considering it a type with semantic restrictions, string just happens to be the internal representation. So it becomes clear a string can't be directly pass to it.

Reply:
This current functionality doesn't work. You can't iterate over the string as you're doing here with the for loop. Must be some unstable part of JS that this is functional.

Do you want me to switch the functionality to one that's consistent with `string`? This might be a breaking change -- which I was trying to avoid

Reply:
Yeah, that's the `String` object, not the `string` primitive. It works, but TS doesn't like it. Is it acceptable to change the code to a traditional for loop?

Reply:
Ah got you! I didn't know TS doesn't like loop on the string primitive. 
>  Is it acceptable to change the code to a traditional for loop?

Yes